spring-data-mongodb/src/test/java/org/springframework/data/mongodb/util/json/ParameterBindingJsonReaderUnitTests.java

```diff
@@ -406,7 +410,7 @@ public void capturingExpressionDependenciesShouldNotThrowParseErrorForSpelOnlyJs
 		String json = "?#{ true ? { 'name': #name } : { 'name' : #name + 'trouble' } }";
 
 		new ParameterBindingDocumentCodec().captureExpressionDependencies(json, (index) -> args[index],
-				new SpelExpressionParser());
+				ValueExpressionParser.create(SpelExpressionParser::new));
```

Comment:
Might make sense to also add some tests to the binding parser and check if the current test setup is making use of value expressions already.